airflow/ui/src/pages/Variables/AddVariableForm.tsx

```diff
@@ -0,0 +1,131 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, Field, HStack, Input, Spacer, Textarea } from "@chakra-ui/react";
+import { useEffect } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { FiSave } from "react-icons/fi";
+
+import { ErrorAlert } from "src/components/ErrorAlert";
+import { Button } from "src/components/ui";
+import { useAddVariable } from "src/queries/useAddVariable";
+
+export type AddVariableBody = {
+  description: string | undefined;
+  key: string;
+  value: string;
+};
+
+type AddVariableFormProps = {
+  readonly onClose: () => void;
+};
+
+const AddVariableForm = ({ onClose }: AddVariableFormProps) => {
+  const { addVariable, error, isPending } = useAddVariable(onClose);
+
+  const {
+    control,
+    formState: { isDirty },
+    handleSubmit,
+    reset,
+    watch,
+  } = useForm<AddVariableBody>({
+    defaultValues: {
+      description: "",
+      key: "",
+      value: "",
+    },
+  });
+
+  const { key, value } = watch();
+
+  useEffect(() => {
+    reset({
+      description: "",
+      key: "",
+      value: "",
+    });
+  }, [reset]);
+
+  const onSubmit = (data: AddVariableBody) => {
+    addVariable(data);
+  };
+
+  return (
+    <>
+      <Controller
+        control={control}
+        name="key"
+        render={({ field }) => (
+          <Field.Root required>
+            <Field.Label fontSize="md">
+              Key <Field.RequiredIndicator />
+            </Field.Label>
+            <Input {...field} required size="sm" />
+          </Field.Root>
+        )}
+      />
+
+      <Controller
+        control={control}
+        name="value"
+        render={({ field }) => (
+          <Field.Root mt={4} required>
+            <Field.Label fontSize="md">
+              Value <Field.RequiredIndicator />
+            </Field.Label>
+            <Textarea {...field} required size="sm" />
+          </Field.Root>
+        )}
+      />
+
+      <Controller
+        control={control}
+        name="description"
+        render={({ field }) => (
+          <Field.Root mb={4} mt={4}>
+            <Field.Label fontSize="md">Description</Field.Label>
+            <Textarea {...field} size="sm" />
+          </Field.Root>
+        )}
+      />
+
+      <ErrorAlert error={error} />
```

Comment:
At the moment if a variable is added and the new one is a conflict, two icons are presented in the error:
![image](https://github.com/user-attachments/assets/3352de51-7eba-472e-8676-3608856e7f80)

Can this be optimized?

Reply:
Ohh, this I am reusing the error alert component, this would be issue with all of them then.
Can we take this in another PR? This has nothing to do with add variables, would be creating problem on other places too.

Reply:
Okay, yeah can be a separate PR.